instrumentation/commons-dbcp2/src/main/java/com/splunk/opentelemetry/commonsdbcp2/DataSourceMetrics.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.opentelemetry.commonsdbcp2;
+
+import com.splunk.opentelemetry.javaagent.bootstrap.GlobalMetricsTags;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.Metrics;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.binder.BaseUnits;
+import java.util.List;
+import java.util.function.Supplier;
+import javax.management.ObjectName;
+import org.apache.commons.dbcp2.BasicDataSourceMXBean;
+
+public final class DataSourceMetrics {
+  public static void registerMetrics(BasicDataSourceMXBean dataSource, ObjectName objectName) {
+    List<Tag> tags = getTags(objectName);
+
+    gauge("db.pool.connections.active", tags, dataSource::getNumActive);
+    gauge("db.pool.connections.idle", tags, dataSource::getNumIdle);
+    gauge("db.pool.connections.min", tags, dataSource::getMinIdle);
+    gauge("db.pool.connections.max", tags, dataSource::getMaxTotal);
```

Comment:
Do we think it might be confusing here to be mixing "min idle" with "max total"?  I think users might not realize the disparity.  Is the goal to track the min and max size of the pool itself?  Maybe we want to consider tracking both "max idle" along with "max total", but give them different names?  So it would end up being something like:

* `db.pool.connections.active`
* `db.pool.connections.idle.num`
* `db.pool.connections.idle.min`
* `db.pool.connections.idle.max`
* `db.pool.connections.capacity.max`

Reply:
> Is the goal to track the min and max size of the pool itself?

Yes, exactly.

Metrics and their naming was inspired by HikariCP built-in metrics instrumentation ([here](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/metrics/micrometer/MicrometerMetricsTracker.java) and [here](https://github.com/brettwooldridge/HikariCP/blob/f9b2c3d372b3f28911a872191ad62953c84bafa4/src/main/java/com/zaxxer/hikari/pool/HikariPool.java#L661)). I tried to measure things that every connection pool exposes (still, it's just a guess) and Hikari does not expose idle.max in its metrics tracker API.

Reply:
But don't you think it's confusing to have min come from idle and max come from total?

Reply:
I thought about this a bit more and I came to a conclusion that `...connections.min` does not make any sense, especially for dbcp2. Idle connection eviction may be disabled (it is in fact disabled by default) and actual min will always = 0, regardless what `idleMin` is actually set to.
You're right, it is confusing when you think about it a bit deeper -- although the min/max (over)simplification is tempting at first.

I changed the names you suggested a bit.

Reply:
Looks good, thanks!